File: memory-domain/src/main/java/com/memory/domain/member/repository/MemberRepositoryCustom.java
package com.memory.domain.member.repository;

import com.memory.domain.member.Member;
import com.memory.domain.member.MemberType;

import java.util.Optional;

public interface MemberRepositoryCustom {

    Optional<Member> findMemberById(Long memberId);

    Optional<Member> findMemberByEmailAndMemberType(String email, MemberType memberType);
}

File: memory-domain/src/main/kotlin/com/memory/domain/member/repository/MemberRepositoryCustom.kt
package com.memory.domain.member.repository

import com.memory.domain.member.Member
import com.memory.domain.member.MemberType
import java.util.Optional

interface MemberRepositoryCustom {

    fun findMemberById(memberId: Long?): Optional<Member>

    fun findMemberByEmailAndMemberType(email: String?, memberType: MemberType?): Optional<Member>
}
